redirect/client.js

```diff
@@ -0,0 +1,54 @@
+(function() {
+
+    function apply(context, template) {
+
+        /**
+         * (window.frameElement || {}).id === "siteEditorFrame" is present in order to prevent redirection from
+         * occurring while in either the Template Editor or Campaign Editor.
+         */
+        const currentPage = window.location.hostname + window.location.pathname.replace(/\/$/, "");
+        const targetPage = context.targetPageUrl.replace(/http(s)?\:\/\//, "");
+        if ((window.frameElement || {}).id === "siteEditorFrame" || currentPage !== targetPage) {
+            return;
+        }
+
+        return new Promise((resolve, reject) => {
+            if ((context.targetPageUrl && context.redirectUrl)
+                && window.location.href !== context.redirectUrl) {
+
+                Evergage.cashDom("body").css("visibility", "hidden");
+
+                Evergage.sendStat({
+                    campaignStats: [
+                        {
+                            control: false,
+                            experienceId: context.experience,
+                            stat: "Impression"
+                        }
+                    ]
+                });
+
+                context.paramsForRedirect = (context.maintainQueryParams && window.location.href.match(/\?.*/))
+                    ? window.location.href.match(/\?.*/)[0]
+                    : "";
+
+                window.location.href = context.redirectUrl + context.paramsForRedirect;
+            }
+        });
```

Comment:
The `window.location.href !== context.redirectUrl` check is unnecessary since you're already doing a check for `currentPage !== targetPage` in the previous `if` statement on line 11. `context.redirectUrl` should just be checked in the `if` statement on line 11 as well. 

Remove this `if` statement (lines 16-17, 36) wrapper entirely, and add a check for `redirectUrl` in the `if` statement on line 11 instead. See my other comment. 

```suggestion
        return new Promise((resolve, reject) => {
            Evergage.cashDom("body").css("visibility", "hidden");

            Evergage.sendStat({
                campaignStats: [
                    {
                        control: false,
                        experienceId: context.experience,
                        stat: "Impression"
                    }
                ]
            });

            context.paramsForRedirect = (context.maintainQueryParams && window.location.href.match(/\?.*/))
                ? window.location.href.match(/\?.*/)[0]
                : "";

            window.location.href = context.redirectUrl + context.paramsForRedirect;
        });
```

Reply:
The check against the redirect URL is still necessary, but yes, it can now be included in the `if` statement. I'm thinking that it should be making the comparison for the `redirectUrl` similarly to the way it's being done for the `targetPageUrl`.